lib/neoboard/widgets/mattermost.ex

```diff
@@ -3,34 +3,55 @@ defmodule Neoboard.Widgets.Mattermost do
   use Neoboard.Pusher
   use Neoboard.Config
   alias Neoboard.TimeService
+  alias Neoboard.Widgets.Mattermost.Authentication
+  alias Neoboard.Widgets.Mattermost.AuthenticationError
   alias Neoboard.Widgets.Mattermost.Fetcher
 
   def start_link do
     {:ok, pid} = GenServer.start_link(__MODULE__, :ok)
-    send(pid, :tick)
-    :timer.send_interval(config[:every], pid, :tick)
+    send(pid, :login)
     {:ok, pid}
   end
 
   def init(:ok) do
-    HTTPoison.start
+    {:ok, %{}}
+  end
 
-    {:ok, %{
+  def handle_info(:login, _) do
+    {:ok, authentication} = Authentication.login(config)
+    send(self(), :tick)
```

Comment:
I am a bit confused by the message flow here.
Does `handle_info(:tick, state)` happen before or after the new state below has been propagated?